File: C#PartTwo/Homework/Arrays/BinarySearch/BinarySearch.cs
namespace BinarySearch
{
    using System;

    class BinarySearch
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            int[] numbers = new int[n];

            for (int i = 0; i < numbers.Length; i++)
            {
                numbers[i] = int.Parse(Console.ReadLine());
            }
            int seekedNumber = int.Parse(Console.ReadLine());

            int startIndex = 0;
            int endIndex = numbers.Length - 1;
            int seekedIndex = -1;
            while (startIndex <= endIndex)
            {
                int middle = (startIndex + endIndex) / 2;

                if (numbers[middle] > seekedNumber)
                {
                    endIndex = middle - 1;
                }
                else if (numbers[middle] < seekedNumber)
                {
                    startIndex = middle + 1;
                }
                else
                {
                    seekedIndex = middle;
                    break;
                }
            }

            Console.WriteLine(seekedIndex);
        }
    }
}

File: JavaScript-Fundamentals/Homeworks/07. Arrays/07. Binary search/BinarySearch.js
function solve(args) {
    var numbers,
        seekedNumber,
        startIndex,
        endIndex,
        middle,
        seekedIndex,
        numbs;

    numbs = args[0].split('\n');
    numbers = [];
    seekedNumber = +numbs[numbs.length - 1];
    for (var i = 1; i < numbs.length - 1; i += 1) {
        numbers.push(+numbs[i]);
    }

    startIndex = 0;
    endIndex = numbers.length - 1;
    seekedIndex = -1;
    while (startIndex <= endIndex) {
        middle = ((startIndex + endIndex) / 2) | 0;

        if (numbers[middle] > seekedNumber) {
            endIndex = middle - 1;
        }
        else if (numbers[middle] < seekedNumber) {
            startIndex = middle + 1;
        }
        else {
            seekedIndex = middle;
            break;
        }
    }

    return seekedIndex;
}